wkcuber/export_wkw_as_tiff.py

```diff
@@ -0,0 +1,171 @@
+        "-d",
+        help="Output directory for the generated dataset.",
+        required=True,
+    )
+
+    parser.add_argument(
+        "--layer_name",
+        "-l",
+        help="Name of the layer that will be converted to a tiff stack",
+        default="color",
+    )
+
+    parser.add_argument("--name", "-n", help="Name of the tiffs", default=None)
+
+    parser.add_argument(
+        "--axis",
+        "-a",
+        help="The axis that the image should be generated along. "
+        "Thus choosing z will print x,y slices.",
+        default="z",
+    )
+
+    parser.add_argument(
+        "--bbox",
+        "-b",
+        help="The BoundingBox of which the tiff stack should be generated.",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--mag", "-m", help="The magnification that should be read", default=1
+    )
+
+    add_verbose_flag(parser)
+    add_distribution_flags(parser)
+
+    return parser
+
+
+def wkw_name_and_bbox_to_tiff_name(
+    name: str, bbox: Dict[str, Tuple[int, int, int]]
+) -> str:
+    return (
+        f'{name}_topleft_{bbox["topleft"][0]}_{bbox["topleft"][1]}_{bbox["topleft"][2]}'
+        f'_size_{bbox["topleft"][0]}_{bbox["topleft"][1]}_{bbox["topleft"][2]}.tiff'
+    )
+
+
+def export_tiff_slice(
+    export_args: Tuple[int, Dict[str, Tuple[int, int, int]], str, str, str, str]
+):
+    slice_number, bbox, dest_path, name, axis, dataset_path = export_args
+    tiff_bbox = bbox
+
+    if axis == "x":
+        tiff_bbox["topleft"] = [
+            tiff_bbox["topleft"][0] + slice_number,
+            tiff_bbox["topleft"][1],
+            tiff_bbox["topleft"][2],
+        ]
+        tiff_bbox["size"] = [1, tiff_bbox["size"][1], tiff_bbox["size"][2]]
+        logging.info(f"starting job: x")
+    if axis == "y":
+        tiff_bbox["topleft"] = [
+            tiff_bbox["topleft"][0],
+            tiff_bbox["topleft"][1] + slice_number,
+            tiff_bbox["topleft"][2],
+        ]
+        tiff_bbox["size"] = [tiff_bbox["size"][0], 1, tiff_bbox["size"][2]]
+    else:
+        tiff_bbox["topleft"] = [
+            tiff_bbox["topleft"][0],
+            tiff_bbox["topleft"][1],
+            tiff_bbox["topleft"][2] + slice_number,
+        ]
+        tiff_bbox["size"] = [tiff_bbox["size"][0], tiff_bbox["size"][1], 1]
+
+    tiff_file_name = wkw_name_and_bbox_to_tiff_name(name, tiff_bbox)
+
+    tiff_file_path = os.path.join(dest_path, tiff_file_name)
+
+    with wkw.Dataset.open(dataset_path) as dataset:
+        tiff_data = dataset.read(tiff_bbox["topleft"], tiff_bbox["size"])
+    tiff_data = np.squeeze(tiff_data)
+    # swap the axis
+    tiff_data.transpose((1, 0))
+
+    logging.info(f"saving slice {slice_number}")
+
+    image = Image.fromarray(tiff_data)
+    image.save(tiff_file_path)
+
+
+def export_tiff_stack(
+    wkw_file_path, wkw_layer, bbox, mag, destination_path, name, axis, args
+):
+    if not os.path.isdir(destination_path):
+        os.mkdir(destination_path)
```

Comment:
You can use `os.makedirs(destination_path, exist_ok=True)` to get rid of the if condition.

Reply:
Nice :+1: